lib/flask_tasks.py

```diff
@@ -51,23 +59,26 @@ def before_first_request():
     def clean_old_tasks():
         logging.debug("Cleanup old tasks")
         """
-        This function cleans up old tasks from our in-memory data structure.
+        This function cleans up old tasks from our in-memory data
+        structure and on disk logs.
         """
-        global tasks
-        while True:
-            # Only keep tasks that are running or that finished less than 5
-            # minutes ago.
-            five_min_ago = timestamp() - 5 * 60
-            tasks = {id: task for id, task in tasks.items()
-                     if 't' not in task or task['t'] > five_min_ago}
-            time.sleep(60)
-
-    if not testing:
-        logging.debug("Production mode, run cleanups")
-        threading.Thread(target=clean_old_tasks)
-        # thread.start()
-    else:
-        logging.debug("Testing mode, no cleanup!")
+        global tasks, stop_cleanup
+        while not stop_cleanup:
+            # Only keep tasks that are running or that finished less than
```

Comment:
the second _that_ is extra: that are running or ~~that~~ finished _in_ less than data_life_time seconds ago.

Reply:
fixed